File: ProyectoCampeonato/src/main/java/org/example/proyectocampeonato/modelo/Combate.java
package org.example.proyectocampeonato.modelo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Combate {

    @EmbeddedId
    private Combate_Id idCombate;

    @Column(nullable = false)
    private String ronda;

    private Integer puntuacionRojo = 0;

    private Integer puntuacionAzul = 0;

    private String estado;

    @ManyToOne
    @JoinColumn(name = "idCompetidor_rojo", nullable = true)
    private Competidor competidorRojo;

    @ManyToOne
    @JoinColumn(name = "idCompetidor_azul", nullable = true)
    private Competidor competidorAzul;

    // Relación con Campeonato_Categoria
    @ManyToOne
    @JoinColumns({
            @JoinColumn(name = "idCampeonato", insertable = false, updatable = false),
            @JoinColumn(name = "idCategoria",  insertable = false, updatable = false)
    })
    private Campeonato_Categoria campeonatoCategoria;
}

File: ProyectoCampeonatoTFG/src/app/interfaces/combate.ts
import { Competidor } from './competidor';
import {Combate_id} from './combate_id';

export interface Combate {
  idCombate: Combate_id;
  ronda: string;
  estado: string;
  puntuacionRojo: number;
  puntuacionAzul: number;
  competidorRojo: Competidor | null;
  competidorAzul: Competidor | null;
}


